pallets/bitacross/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self { admin: None }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			if let Some(ref admin) = self.admin {
+				Admin::<T>::put(admin);
+			}
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Set the admin account
+		///
+		/// Weights should be 2 DB writes: 1 for mode and 1 for event
+		#[pallet::call_index(0)]
+		#[pallet::weight((2 * T::DbWeight::get().write, DispatchClass::Normal, Pays::No))]
+		pub fn set_admin(
+			origin: OriginFor<T>,
+			new_admin: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::SetAdminOrigin::ensure_origin(origin)?;
+			Admin::<T>::put(new_admin.clone());
+			Self::deposit_event(Event::AdminSet { new_admin: Some(new_admin) });
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight({195_000_000})]
+		pub fn add_relayer(origin: OriginFor<T>, account: Identity) -> DispatchResult {
+			Self::ensure_admin_or_root(origin)?;
+			ensure!(account.is_substrate() || account.is_evm(), Error::<T>::UnsupportedRelayerType);
+			// we don't care if `account` already exists
+			Relayer::<T>::insert(account.clone(), ());
+			Self::deposit_event(Event::RelayerAdded { who: account });
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight({195_000_000})]
+		pub fn remove_relayer(origin: OriginFor<T>, account: Identity) -> DispatchResult {
+			Self::ensure_admin_or_root(origin)?;
+			ensure!(Relayer::<T>::contains_key(&account), Error::<T>::RelayerNotExist);
+			Relayer::<T>::remove(account.clone());
+			Self::deposit_event(Event::RelayerRemoved { who: account });
+			Ok(())
+		}
+
+		/// ---------------------------------------------------
+		/// The following extrinsics are supposed to be called by TEE only
+		/// ---------------------------------------------------
+		#[pallet::call_index(30)]
+		#[pallet::weight(({195_000_000}, DispatchClass::Normal, Pays::No))]
+		pub fn btc_wallet_generated(
+			origin: OriginFor<T>,
+			address: Address33,
+		) -> DispatchResultWithPostInfo {
+			let tee_account = T::TEECallOrigin::ensure_origin(origin)?;
+			Vault::<T>::try_mutate(tee_account, |v| {
+				ensure!(!v.has_btc(), Error::<T>::BtcWalletAlreadyExist);
+				v.btc = Some(address);
+				Self::deposit_event(Event::BtcWalletGenerated { address });
+				Ok(Pays::No.into())
+			})
+		}
+
+		#[pallet::call_index(31)]
+		#[pallet::weight(({195_000_000}, DispatchClass::Normal, Pays::No))]
+		pub fn eth_wallet_generated(
+			origin: OriginFor<T>,
+			address: Address20,
+		) -> DispatchResultWithPostInfo {
+			let tee_account = T::TEECallOrigin::ensure_origin(origin)?;
+			Vault::<T>::try_mutate(tee_account, |v| {
+				ensure!(!v.has_eth(), Error::<T>::EthWalletAlreadyExist);
+				v.eth = Some(address);
+				Self::deposit_event(Event::EthWalletGenerated { address });
+				Ok(Pays::No.into())
+			})
+		}
+
+		// TODO: placeholder
```

Comment:
What do you plan to do regarding this TODO ? :)

Reply:
It should be the callback extrinsic of the signing task. Currently it's an empty impl:
```
pub fn task_complete(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
	let _ = T::TEECallOrigin::ensure_origin(origin)?;
	Ok(Pays::No.into())
}
```

or something like `btc_signing_complete` and `eth_signing_complete` ... I'm not sure about the parameter tho, need to be further specified by bc colleagues - cc @wangminqi 

But we can have some preliminary implementation, like request-identifier (H256?), singature itself, input payload hash or something

Reply:
Is it a good idea to include the signature anywhere, since it could be used by anyone, I don't think it is immediately being used for any transactions. So do we want to use extrinsics only for logging on the parachain side?

Reply:
AFAICT it's fine for the signature to be public (which implies it can be used by anyone) - SC will check the validity/expiration of these.

Either way, it's easy to show/hide such information in the extrinsic parameter